services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/UnicastGoalStateV2.java

```diff
@@ -0,0 +1,100 @@
+    including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and / or sell copies of the Software, and to permit persons
+    to whom the Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+package com.futurewei.alcor.dataplane.entity;
+
+import com.futurewei.alcor.schema.Goalstate.GoalStateV2;
+import com.futurewei.alcor.web.entity.dataplane.UnicastGoalStateByte;
+
+public class UnicastGoalStateV2 {
+    private String hostIp;
+    private String vpcId;
+    private String topic;
+    private String subTopic;
+
+    private GoalStateV2 goalState;
+    private GoalStateV2.Builder goalStateBuilder;
+
+    public UnicastGoalStateV2() {
+        goalStateBuilder = GoalStateV2.newBuilder();
+    }
+
+    public UnicastGoalStateV2(String hostIp, GoalStateV2 goalState) {
+        this.hostIp = hostIp;
+        this.goalState = goalState;
+    }
+
+    public UnicastGoalStateV2(String hostIp, String vpcId, GoalStateV2 goalState) {
+        this.hostIp = hostIp;
+        this.vpcId = vpcId;
+        this.goalState = goalState;
+    }
+
+    public String getHostIp() {
+        return hostIp;
+    }
+
+    public void setHostIp(String hostIp) {
+        this.hostIp = hostIp;
+    }
+
+    public String getVpcId() {
+        return vpcId;
+    }
+
+    public void setVpcId(String vpcId) {
+        this.vpcId = vpcId;
+    }
+
+    public String getTopic() {
+        return topic;
+    }
+
+    public void setTopic(String topic) {
+        this.topic = topic;
+    }
+
+    public String getSubTopic() {
+        return subTopic;
+    }
+
+    public void setSubTopic(String subTopic) {
+        this.subTopic = subTopic;
+    }
+
+    public GoalStateV2 getGoalState() {
+        return goalState;
+    }
+
+    public void setGoalState(GoalStateV2 goalState) {
+        this.goalState = goalState;
+    }
+
+    public GoalStateV2.Builder getGoalStateBuilder() {
+        return goalStateBuilder;
+    }
+
+    public void setGoalStateBuilder(GoalStateV2.Builder goalStateBuilder) {
+        this.goalStateBuilder = goalStateBuilder;
+    }
+
+    public UnicastGoalStateByte getUnicastGoalStateByte() {
+        UnicastGoalStateByte unicastGoalStateByte = new UnicastGoalStateByte();
+        unicastGoalStateByte.setNextTopic(this.topic);
```

Comment:
@VanderChen We would need to setNextSubTopic as well, right?

Reply:
This involves another question, whether use Pulsar worker to forward messages. This design was originally used to solve the multiple sending problem of multicastGS. DPM only needs to send GS once, and the worker will complete it multiple times. To be consistent, UnicastGS also uses the worker forwarding method.
i.e. 
DPM(UniWorkerTopic, (NextTopic, GS)) -- (NextTopic, GS)--> Pulsar Worker(NextTopic, GS) --(GS)--> ACA
DPM(MultiWorkerTopic, (List<NextTopic>, GS)) --(List<NextTopic>, GS)--> Pulsar Worker(Traversal_NextTopic, GS) --(GS)--> ACA
Thus, considering compatibility with the current version and not sending more information to the worker, I did not add it. I'm not sure if this is the right thing to do.
There are two options for those containing SubTopic. One is not to use workers, this does not require the UnicastGoalStateByte class. The other is to deploy new Pulsar Workers. I prefer the former one.
i.e.
DPM((Topic, SubTopic), GS) --(GS)--> ACA

Reply:
I am not quite sure what's problem here. Where is the Pulsar worker located? In the DPM (such as pulsarClient in the pulsar\DataPlaneClientImpl.java) or somewhere else?
Does your prefer way to eliminate Pulsar worker and rely on the GRPC's unicast and multicast goalstates?

Reply:
Do you have this problem resolved? Let us discuss it in today's open-source meeting.